File: doc-examples/example-jsonb-kotlin/src/main/kotlin/example/ProgrammaticJsonbConfigFactory.kt
package example

import io.micronaut.context.annotation.Factory
import io.micronaut.context.annotation.Requires
import jakarta.inject.Singleton
import jakarta.json.bind.JsonbConfig

@Factory
@Requires(property = "spec.name", value = "jsonb-programmatic-config")
class ProgrammaticJsonbConfigFactory {
    @Singleton
    fun jsonbConfig(): JsonbConfig = JsonbConfig()
        .withSerializers(ProgrammaticCodeSerializer())
        .withDeserializers(ProgrammaticCodeDeserializer())
}

File: doc-examples/example-jsonb-python/src/main/python/example/ProgrammaticJsonbConfigFactory.py
from jakarta.inject import Singleton
from jakarta.json.bind import JsonbConfig
from micronaut.context.annotation import Factory, Requires

from example.ProgrammaticCodeDeserializer import ProgrammaticCodeDeserializer
from example.ProgrammaticCodeSerializer import ProgrammaticCodeSerializer


@Factory
@Requires(property="spec.name", value="jsonb-programmatic-config")
class ProgrammaticJsonbConfigFactory:
    @Singleton
    def jsonb_config(self) -> JsonbConfig:
        return (
            JsonbConfig()
            .withSerializers(ProgrammaticCodeSerializer())
            .withDeserializers(ProgrammaticCodeDeserializer())
        )
